lib/rust/parser/generate-java/java/org/enso/syntax2/Parser.java

```diff
@@ -131,7 +133,13 @@ public Tree parse(CharSequence input) {
     var metadata = getMetadata(state);
     serializedTree.order(ByteOrder.LITTLE_ENDIAN);
     var message = new Message(serializedTree, input, base, metadata);
-    return Tree.deserialize(message);
+    try {
+      return Tree.deserialize(message);
+    } catch (BufferUnderflowException | IllegalArgumentException e) {
+      LoggerFactory.getLogger(this.getClass())
+          .error("Unrecoverable parser failure for: {}", input, e);
```

Comment:
This code is calling [error(Object,Throwable)](https://www.slf4j.org/apidocs-2.1.0/org/slf4j/Logger.html#error(java.lang.String,java.lang.Throwable)) method. What's its behavior in Enso?

#### When running `runtime-integration-tests`

Will it print the message as well as exception to the (CI) console?

#### When running in `enso` CLI mode

Will it print the message as well as exception to the (CI) console? Will it do something different?

#### When running in Enso IDE mode

What will appear in console and what will appear in associated log file?

Reply:
This PR changes nothing when it comes to how log messages are consumed, this is calling a plain slf4j api. The actual printing is dependent on the implementation/configuration, as you know.

If you don't like the format/output of the logger in CLI or tests then please create a ticket for that. IDE works as expected.